packages/ui/src/views/pull-request/pull-request-list.tsx

```diff
@@ -0,0 +1,90 @@
+import { useMemo, useState } from 'react'
+import { Link } from 'react-router-dom'
+
+import * as StackedList from '../../components/stacked-list'
+import { PullRequestListDescription } from './pull-request-list-description'
+import { PullRequestListHeaderTitle } from './pull-request-list-header-title'
+import { PullRequestListTitle } from './pull-request-list-title'
+import { PullRequestType } from './types'
+
+export interface PullRequestListPageProps {
+  pullRequests?: PullRequestType[]
+  // LinkComponent: React.ComponentType<{ to: string; children: React.ReactNode }>
+  handleResetFilters?: () => void
+  hasActiveFilters?: boolean
+  query?: string
+  handleResetQuery: () => void
+  closed_prs?: number
+  open_prs?: number
+}
+
+export function PullRequestList({ pullRequests, open_prs, closed_prs }: PullRequestListPageProps) {
+  const [headerFilter, setHeaderFilter] = useState('open')
+  const filteredData = useMemo<PullRequestType[]>(() => {
+    if (!pullRequests) return []
+
+    return pullRequests.filter(pr => {
+      if (headerFilter === 'open') return pr.state === 'open'
+      if (headerFilter === 'closed') return pr.state !== 'open' || pr.merged !== null
+      return true
+    })
+  }, [headerFilter])
+
+  return (
+    <>
+      {!!filteredData?.length && (
```

Comment:
Looks like you can just return early instead of testing in the JSX.
```suggestion
  if (!filteredData?.length) return null

  return (
```

Not a blocker, just will make the component 2% easier to reason about.

Reply:
done